docs/source/topics/testing.rst

```diff
@@ -0,0 +1,310 @@
+Testing
+=======
+
+Chalice provides a :ref:`test client <testing-api>` in ``chalice.test`` that
+you can use to test your Chalice applications.  This client lets you invoke
+Lambda function and event handlers directly, as well as test your REST APIs.
+
+Lambda Functions
+----------------
+
+To test lambda functions, use the
+:meth:`Client.lambda_.invoke <TestLambdaClient.invoke>` method.  The
+test client is intended to be used as a context manager.  For example,
+given this sample app:
+
+
+.. code-block:: python
+
+   from chalice import Chalice
+
+   app = Chalice(app_name="testclient")
+
+   @app.lambda_function()
+   def foo(event, context):
+       return {'hello': 'world'}
+
+   @app.lambda_function()
+   def bar(event, context):
+       return {'event': event}
+
+
+Here's how you can test these functions with the test client:
```

Comment:
What is the file path for these tests? Is it under a `test_app.py` file? I'm wondering if this guide should help suggest where to write tests as well. I also wonder if we help give direction on how to write tests in a separate directory from the `app.py` file in the case they want a `tests` directory.

Reply:
I can update this.  It doesn't really matter where you put them, either a top level `test_app.py` or a `tests/test_app.py`.  I'll update the docs to recommend a `tests/test_app.py` along with instructions on how you might run these tests with pytest.